index.js

```diff
@@ -339,24 +292,85 @@ module.exports = {
   TYPE_REJECT,
   TYPE_MESSAGE,
 
+  typeToString,
+
   MIME_APPLICATION_OCTET_STREAM,
   MIME_TEXT_PLAIN_UTF8,
   MIME_APPLICATION_JSON,
 
-  serializeAck,
-  serializeResponse,
-  serializeError,
-  serializePrepare,
-  serializeFulfill,
-  serializeReject,
-  serializeMessage,
-
-  deserializeAck,
-  deserializeResponse,
-  deserializeError,
-  deserializePrepare,
-  deserializeFulfill,
-  deserializeReject,
-  deserializeMessage,
-  deserializeClpPacket
+  serialize,
+  deserialize,
+
+  // The following legacy functions use an alternative format to access the exposed
```

Comment:
I wouldn't call these legacy. If the parameters being passed in are different between the types, they should be separate functions rather than just using a flag to determine which function should be run